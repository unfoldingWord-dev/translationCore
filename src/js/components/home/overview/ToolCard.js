import React, { Component } from 'react'
import TemplateCard from './TemplateCard'

class ToolCard extends Component {

  heading(callback) {
    let link = this.content() ? <a onClick={callback}>Change Tool</a> : <a></a>
    return (
      <span>Current Tool {link}</span>
    );
  }

  content() {
    let content;
    let { toolName } = this.props.reducers.toolsReducer;
    if (toolName) {
      content = (
        <div>
          <strong>{toolName}</strong>
          <p>details</p>
        </div>
      );
    }
    return content;
  }

  disabled() {
    let { projectSaveLocation } = this.props.reducers.projectDetailsReducer;
    return !projectSaveLocation;
  }

  render() {
    let emptyMessage = 'Select a tool';
    let emptyButtonLabel = 'Tool';
    let emptyButtonOnClick = () => { this.props.actions.goToNextStep() };
    return (
      <TemplateCard
        heading={this.heading(emptyButtonOnClick)}
        content={this.content()}
        emptyMessage={emptyMessage}
        emptyButtonLabel={emptyButtonLabel}
        emptyButtonOnClick={emptyButtonOnClick}
        disabled={this.disabled()}
      />
    )
  }
}

export default ToolCard
