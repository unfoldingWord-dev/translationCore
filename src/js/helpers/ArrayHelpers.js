/**
 * @description - Groups consecutive numbers in an array
 * @param {Array} numbers - array of numbers to be grouped
 * @returns {Array} - grouped array of array of consecutive numbers
 */
export const groupConsecutiveNumbers = (numbers) => (
  numbers.reduce(function(accumulator, currentValue, currentIndex, originalArray) {
    if (currentValue) { // ignore undefined entries
      // if this iteration is consecutive to the last, add it to the previous run
      if (currentValue - originalArray[currentIndex - 1] === 1) {
        accumulator[accumulator.length - 1].push(currentValue);
      } else { // the start of a new run including first element
        // create a new subarray with this as the start
        accumulator.push([currentValue]);
      }
    }
    return accumulator; // return state for next iteration
  }, [])
);
/**
 * @description - Deletes indices from an array safely
 * @param {Array} array - array elements to delete from
 * @returns {Array} - the resulting array after indexes were safely removed
 */
export const deleteIndices = (array, indices) => {
  let _array = JSON.parse(JSON.stringify(array));
  indices.sort( (a,b) => b - a );
  indices.forEach(index => {
    if (index >= 0) _array.splice(index, 1);
  });
  return _array;
};
