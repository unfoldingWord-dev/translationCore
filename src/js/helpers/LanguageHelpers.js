/* eslint-disable no-console */

let languages = null; // cache languages for speed up
let languageListByName = null; // list caching for speed up

/**
 * @description - returns a list of language objects from langnames.json sorted by language code.
 * @return {array}
 */
export const getLanguages = () => {
  if (!languages) {
    const langList = require('../../assets/langnames');
    languages = [];
    for (let language of langList) {
      const name = language.ln || language.ang || language.lc;
      if (language.lc) {
        languages.push({code: language.lc, name: name, ltr: language.ld !== 'rtl',
          namePrompt: name + ' [' + language.lc + ']'});
      }
    }
    languages.sort(function(a,b) {return (a.code > b.code) ? 1 : ((b.code > a.code) ? -1 : 0) } );
  }
  return languages;
};

/**
 * @description - searches language array to match language code.
 * @param {string} code
 * @return {object} language entry matched or null if no match
 */
export const getLanguageByCode = (code) => {
  const languageList = getLanguages();
  for (let language of languageList) {
    if (language.code === code) {
      return language;
    }
  }
  return null;
};

/**
 * @description - searches language array to do case insensitive match of language name.
 * @param {string} name
 * @return {array} language entry matched or null if no match
 */
export const getLanguageByName = (name) => {
  const languageList = getLanguages();
  const nameLC = name.toLowerCase();
  for (let language of languageList) {
    if (language.name.toLowerCase() === nameLC) {
      return language;
    }
  }
  return null;
};

/**
 * @description verify language code
 * @param {string} languageID - language code
 * @return {boolean} true if valid code
 */
export const isLanguageCodeValid = (languageID) => {
  const language = getLanguageByCode(languageID);
  return language != null;
};

/**
 * @description verify language name (case insensitive)
 * @param {string} languageName - language name
 * @return {boolean} true if valid code
 */
export const isLanguageNameValid = (languageName) => {
  const language = getLanguageByName(languageName);
  return language != null;
};


/**
 * @description - returns cached list of languages sorted by name.  And if cache is empty,
 *                  it is filled from list of languages and sorts them by language name
 * @return {*}
 */
export const getLanguagesSortedByName = () => {
  if (!languageListByName) {
    languageListByName = getLanguages().slice(0); // clone list
    languageListByName.sort(function (a, b) { // sort by language name
      return (a.name > b.name) ? 1 : ((b.name > a.name) ? -1 : 0);
    });
  }
  return languageListByName;
};
