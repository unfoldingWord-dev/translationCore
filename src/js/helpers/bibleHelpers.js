/* eslint-disable no-console */
import Path from 'path-extra';
import fs from 'fs-extra';
import BooksOfBible from '../../../tcResources/books';
import {getLatestVersionInPath} from "./ResourcesHelpers";

/**
 *
 * @param {string} bookAbbr - The book abbreviation to convert
 */
export function convertToFullBookName(bookAbbr) {
  if (!bookAbbr) return;
  return BooksOfBible[bookAbbr.toString().toLowerCase()];
}

/**
 *
 * @param {string} projectBook - Book abbreviation
 */
export function isOldTestament(projectBook) {
  var passedBook = false;
  for (var book in BooksOfBible) {
    if (book === projectBook) passedBook = true;
    if (BooksOfBible[book] === 'Malachi' && passedBook) {
      return true;
    }
  }
  return false;
}

/**
 * determine
 * @param bookId
 * @return {{resourceLanguage: string, bibleID: string}}
 */
export function getOLforBook(bookId) {
  const isOT = isOldTestament(bookId);
  const languageId = (isOT) ? 'he' : 'grc';
  const bibleId = (isOT) ? 'uhb' : 'ugnt';
  return {languageId, bibleId};
}

/**
 * This checks if a project is missing verses.
 * @param projectDir
 * @param bookId
 * @param resourceDir
 * @return {boolean}
 */
export const isProjectMissingVerses = (projectDir, bookId, resourceDir) => {
  try {
    let languageId = 'en';
    const resourcePath = Path.join(resourceDir, languageId, 'bibles', 'ult');
    const versionPath = getLatestVersionInPath(resourcePath) || resourcePath;
    const indexLocation = Path.join(versionPath, 'index.json');
    const expectedVerses = fs.readJSONSync(indexLocation);
    const actualVersesObject = {};
    const currentFolderChapters = fs.readdirSync(Path.join(projectDir, bookId));
    let chapterLength = 0;
    for (let currentChapterFile of currentFolderChapters) {
      let currentChapter = Path.parse(currentChapterFile).name;
      if (!parseInt(currentChapter)) continue;
      chapterLength++;
      let verseLength = 0;
      try {
        let currentChapterObject = fs.readJSONSync(
          Path.join(projectDir, bookId, currentChapterFile)
        );
        for (let verseIndex in currentChapterObject) {
          let verse = currentChapterObject[verseIndex];
          if (verse && verseIndex > 0) verseLength++;
        }
      } catch (e) {
        console.warn(e);
      }
      actualVersesObject[currentChapter] = verseLength;
    }
    actualVersesObject.chapters = chapterLength;
    let currentExpectedVerese = expectedVerses[bookId];
    return (
      JSON.stringify(currentExpectedVerese) !==
      JSON.stringify(actualVersesObject)
    );
  } catch (e) {
    console.warn(
      'ult index file not found missing verse detection is invalid. Please delete ~/translationCore/resources folder'
    );
    return false;
  }
};
