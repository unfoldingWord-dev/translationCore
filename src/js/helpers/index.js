import { generateTimestamp } from './TimestampGenerator.js';
import * as loadHelpers from './LoadHelpers.js';

export {
  generateTimestamp,
  loadHelpers,
};
