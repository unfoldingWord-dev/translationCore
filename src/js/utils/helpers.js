import Path from 'path';
import * as fs from 'fs-extra';
import ManifestGenerator from '../components/core/create_project/ProjectManifest';
import BooksOfBible from '../components/core/BooksOfBible';

module.exports.isOldTestament = function (projectBook) {
    var passedBook = false;
    for (var book in BooksOfBible) {
        if (book == projectBook) passedBook = true;
        if (BooksOfBible[book] == "Malachi" && passedBook) {
            return true;
        }
    }
    return false;
}

module.exports.checkIfValidBetaProject = function (manifest) {
    if (manifest && manifest.project) return manifest.project.id == "eph" || manifest.project.id == "tit";
    else if (manifest && manifest.ts_project) return manifest.ts_project.id == "eph" || manifest.ts_project.id == "tit";
}

/**
 * 
 * @param {*} path 
 * @param {*} file 
 */
export function loadFile(path, file) {
    try {
        var manifest = fs.readJsonSync(Path.join(path, file));
        return manifest;
    }
    catch (e) {
        return null
    }
}

/**
 * @description - Generates and saves a translationCore manifest file
 * @param {string} projectSaveLocation - Filepath of where the translationCore manifest file will
 * be saved. Must be an ABSOLUTE PATH
 * @param {object} data - The translationCore manifest data to be saved
 * @param {object} tsManifest - The translationStudio manifest data loaded from a translation
 * studio project
 */
module.exports.saveManifest = function (projectSaveLocation, link, oldManifest, currentUser) {
    var data = {
        user: [currentUser],
        repo: link
    }
    var manifest;
    try {
        var manifestLocation = Path.join(projectSaveLocation, 'manifest.json');
        if (oldManifest.package_version == '3') {
            //some older versions of ts-manifest have to be tweaked to work
            manifest = this.fixManifestVerThree(oldManifest);
        } else {
            manifest = ManifestGenerator(data, oldManifest);
        }
        fs.outputJsonSync(manifestLocation, manifest);
    }
    catch (err) {
        console.error(err);
    }
    return manifest;
}

/**
 * @desription - Uses the tc-standard format for projects to make package_version 3 compatible
 * @param oldManifest - The name of an employee.
 */
module.exports.fixManifestVerThree = function (oldManifest) {
    var newManifest = {};
    try {
        for (var oldElements in oldManifest) {
            newManifest[oldElements] = oldManifest[oldElements];
        }
        newManifest.finished_chunks = oldManifest.finished_frames;
        newManifest.ts_project = {};
        newManifest.ts_project.id = oldManifest.project_id;
        newManifest.ts_project.name = this.convertToFullBookName(oldManifest.project_id);
        for (var el in oldManifest.source_translations) {
            newManifest.source_translations = oldManifest.source_translations[el];
            var parameters = el.split("-");
            newManifest.source_translations.language_id = parameters[1];
            newManifest.source_translations.resource_id = parameters[2];
            break;
        }
    } catch (e) {
        console.error(e);
    }
    return newManifest;
}

module.exports.convertToFullBookName = function (bookAbbr) {
    if (!bookAbbr) return;
    return BooksOfBible[bookAbbr.toString().toLowerCase()];
}