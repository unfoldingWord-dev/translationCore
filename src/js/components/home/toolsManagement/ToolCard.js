import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Card, CardHeader } from 'material-ui';
import { Glyphicon } from 'react-bootstrap';
import { connect } from 'react-redux';
import _ from 'lodash';
// helpers
import * as ToolCardHelpers from '../../../helpers/ToolCardHelpers';
import { getTranslation } from '../../../helpers/localizationHelpers';
import { getGatewayLanguageList, hasValidOL } from '../../../helpers/gatewayLanguageHelpers';
// components
import Hint from '../../Hint';
import {
  getProjectBookId,
  getSetting,
  getSelectedToolName,
} from '../../../selectors';
import {
  WORD_ALIGNMENT, TRANSLATION_WORDS, TRANSLATION_NOTES,
} from '../../../common/constants';
import ToolCardBoxes from './ToolCardBoxes';
import ToolCardProgress from './ToolCardProgress';
import GlDropDownList from './GlDropDownList';
import ToolCardNotificationBadges from './ToolCardNotificationBadges';
// selectors
// consts

class ToolCard extends Component {
  constructor(props) {
    super(props);
    this.selectionChange = this.selectionChange.bind(this);
    this.handleSelect = this.handleSelect.bind(this);
    this.onLaunchClick = this.onLaunchClick.bind(this);
    this.loadProgress = _.debounce(this.loadProgress.bind(this), 200);
    this.state = {
      showDescription: false,
      progress: 0,
      selectedCategoriesChanged: false,
      glSelectedChanged: false,
      selectedGL: '',
      gatewayLanguageList: null,
    };
  }

  componentWillMount() {
    let {
      tool,
      bookId,
      glSelected: selectedGL,
    } = this.props;
    const gatewayLanguageList = getGatewayLanguageList(bookId, tool.name);

    // if there is only one gateway Language then select it as the GL for the tool card.
    if (gatewayLanguageList.length === 1) {
      selectedGL = gatewayLanguageList[0].code;
    }
    this.selectionChange(selectedGL);
    this.setState({ selectedGL, gatewayLanguageList });
  }

  componentDidMount() {
    this.loadProgress();
  }

  componentDidUpdate(prevProps) {
    const glSelectedChanged = prevProps.glSelected !== this.props.glSelected;

    if (glSelectedChanged) {
      this.setState({ glSelectedChanged });
    }

    if (!_.isEqual(prevProps.selectedCategories, this.props.selectedCategories)) {
      this.setState({ selectedCategoriesChanged: true });
      this.loadProgress();
    }
  }

  loadProgress() {
    const { tool, actions } = this.props;
    const { progress } = this.state;

    setTimeout(() => {
      const results = {};
      // TODO: This is an antipattern. Should update the state and use the prop coming from the state instead of assigning the result to an argument.
      actions.getProjectProgressForTools(tool.name, results);
      const toolProgress = results.progress;

      if (progress !== toolProgress) {
        this.setState({ progress: toolProgress ? toolProgress : 0 });
      }
    }, 0);
  }

  selectionChange(selectedGL) {
    if (selectedGL && selectedGL.trim()) {
      this.props.actions.setProjectToolGL(this.props.tool.name, selectedGL);
      this.setState({ selectedGL });
      this.loadProgress();
    }
  }

  getLaunchDisableMessage(id, developerMode, translate, name, selectedCategories) {
    const toolsWithCategories = [TRANSLATION_WORDS , TRANSLATION_NOTES];
    let launchDisableMessage = ToolCardHelpers.getToolCardLaunchStatus(this.state.selectedGL, id, developerMode, translate);

    if (!launchDisableMessage) { // if no errors, make sure we have original language
      const olBookPath = hasValidOL(id);

      if (!olBookPath) {
        launchDisableMessage = translate('tools.book_not_supported');
      }
    }

    if (!launchDisableMessage && !developerMode) { // if no errors and not developer mode , make sure we have a gateway language
      const { gatewayLanguageList } = this.state;
      launchDisableMessage = (gatewayLanguageList && gatewayLanguageList.length) ? null : translate('tools.book_not_supported');
    }

    if (!launchDisableMessage && (toolsWithCategories.includes(name) && selectedCategories.length === 0)) {
      launchDisableMessage = translate('tools.no_checks_selected');
    }

    return launchDisableMessage;
  }

  /**
   * Handles selecting the tool
   */
  handleSelect() {
    const { onSelect, tool } = this.props;
    onSelect(tool.name);
  }

  onLaunchClick() {
    const {
      isOLBookVersionMissing,
      toggleHomeView,
      selectedToolName,
      tool,
      actions: { warnOnInvalidations },
    } = this.props;
    const { selectedCategoriesChanged, glSelectedChanged } = this.state;
    const newSelectedToolName = tool.name;

    if (isOLBookVersionMissing) {
      // Show dialog with option to download missing resource
      this.props.onMissingResource();
    } else if (selectedToolName && !glSelectedChanged && !selectedCategoriesChanged && (selectedToolName === newSelectedToolName)) {
      // Show tool (Without loading tool data)
      toggleHomeView(false);
      warnOnInvalidations(newSelectedToolName);
    } else {
      // Load tool data then show tool
      this.handleSelect();
    }
  }

  render() {
    const {
      tool,
      bookId,
      translate,
      developerMode,
      actions: {
        updateCategorySelection,
        updateSubcategorySelection,
      },
      selectedCategories,
      availableCategories,
    } = this.props;
    const {
      progress, selectedGL, gatewayLanguageList,
    } = this.state;
    const launchDisableMessage = this.getLaunchDisableMessage(bookId, developerMode, translate, tool.name, selectedCategories);
    let desc_key = null;
    let showCheckBoxes = false;

    switch (tool.name) {
    case WORD_ALIGNMENT:
      desc_key = 'tools.alignment_description';
      break;

    case TRANSLATION_WORDS:
      showCheckBoxes = true;
      desc_key = 'tools.tw_part1_description';
      break;

    case TRANSLATION_NOTES:
      showCheckBoxes = true;
      break;

    default:
      break;
    }

    let descriptionLocalized = tool.description;

    if (desc_key) {
      descriptionLocalized = getTranslation(translate, desc_key, tool.description);
    }

    return (
      <MuiThemeProvider>
        <Card style={{ margin: '6px 0px 10px' }}>
          <img
            alt={tool.name}
            style={{
              float: 'left', height: '90px', margin: '10px',
            }}
            src={tool.badge}
          />
          <CardHeader
            title={tool.title}
            titleStyle={{ fontWeight: 'bold' }}
            subtitle={tool.version}
            style={{ display: 'flex', justifyContent: 'space-between' }}>
            <ToolCardNotificationBadges tool={tool} translate={translate} selectedCategories={selectedCategories} />
          </CardHeader><br />
          <ToolCardProgress progress={progress} />
          {
            showCheckBoxes &&
            <ToolCardBoxes
              key={selectedGL}
              toolName={tool.name}
              selectedCategories={selectedCategories}
              availableCategories={availableCategories}
              onCategoryChecked={updateCategorySelection}
              onSubcategoryChecked={updateSubcategorySelection}
              bookId={bookId}
              translate={translate}
              selectedGL={selectedGL}
            />
          }
          {this.state.showDescription ?
            (<div>
              <span style={{
                fontWeight: 'bold', fontSize: '16px', margin: '0px 10px 10px',
              }}>{translate('tools.description')}</span>
              <p style={{ padding: '10px' }}>
                {descriptionLocalized}
              </p>
            </div>) : (<div />)
          }
          <div style={{
            display: 'flex', justifyContent: 'space-between', flexDirection: 'row', alignItems: 'center',
          }}>
            <div style={{ display: 'flex', justifyContent: 'space-between' }}>
              <div
                style={{
                  padding: '10px 10px 0px', fontSize: '18px', cursor: 'pointer',
                }}
                onClick={() => this.setState({ showDescription: !this.state.showDescription })}
              >
                <span>{this.state.showDescription ? translate('tools.see_less') : translate('tools.see_more')}</span>
                <Glyphicon
                  style={{ fontSize: '18px', margin: '0px 0px 0px 6px' }}
                  glyph={this.state.showDescription ? 'chevron-up' : 'chevron-down'}
                />
              </div>
            </div>
            <GlDropDownList
              translate={translate}
              selectedGL={this.state.selectedGL}
              selectionChange={this.selectionChange}
              gatewayLanguageList={gatewayLanguageList}
            />
            <Hint
              position={'left'}
              size='medium'
              label={launchDisableMessage}
              enabled={launchDisableMessage ? true : false}
            >
              <button
                disabled={launchDisableMessage ? true : false}
                className='btn-prime'
                onClick={this.onLaunchClick}
                style={{ width: '90px', margin: '10px' }}
              >
                {translate('buttons.launch_button')}
              </button>
            </Hint>
          </div>
        </Card>
      </MuiThemeProvider>
    );
  }
}

ToolCard.propTypes = {
  onSelect: PropTypes.func.isRequired,
  translate: PropTypes.func.isRequired,
  bookId: PropTypes.string.isRequired,
  tool: PropTypes.object.isRequired,
  developerMode: PropTypes.bool.isRequired,
  actions: PropTypes.shape({
    getProjectProgressForTools: PropTypes.func.isRequired,
    setProjectToolGL: PropTypes.func.isRequired,
    updateSubcategorySelection: PropTypes.func.isRequired,
    updateCategorySelection: PropTypes.func.isRequired,
    warnOnInvalidations: PropTypes.func.isRequired,
  }),
  selectedCategories: PropTypes.array.isRequired,
  availableCategories: PropTypes.object.isRequired,
  isOLBookVersionMissing: PropTypes.bool.isRequired,
  onMissingResource: PropTypes.func.isRequired,
  selectedToolName: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]),
  toggleHomeView: PropTypes.func.isRequired,
  glSelected: PropTypes.string.isRequired,
};

ToolCard.contextTypes = { store: PropTypes.any };

const mapStateToProps = (state) => ({
  bookId: getProjectBookId(state),
  developerMode: getSetting(state, 'developerMode'),
  selectedToolName: getSelectedToolName(state),
});

export default connect(mapStateToProps)(ToolCard);
