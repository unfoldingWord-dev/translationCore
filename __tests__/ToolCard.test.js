/* eslint-env jest */
import React from 'react';
import ToolCard from '../src/js/components/home/toolsManagement/ToolCard';
import { DEFAULT_GATEWAY_LANGUAGE } from '../src/js/helpers/LanguageHelpers';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { shallow } from 'enzyme';

jest.mock('../src/js/components/home/toolsManagement/ToolCardProgress', () => 'ToolCardProgress');
jest.mock('../src/js/components/home/toolsManagement/GlDropDownList', () => 'GlDropDownList');

// Tests for ToolCard React Component
describe('Test ToolCard component',()=>{
  test('Comparing ToolCard Component render with snapshot taken 11/07/2017 in __snapshots__ should match', () => {
    const props = {
      loggedInUser: true,
      currentProjectToolsProgress: {
        testTool: 0.5
      },
      currentProjectToolsSelectedGL: {
        testTool: 'en'
      },
      manifest: {
        project: {
          id: 'tit'
        }
      },
      metadata: {
        name: 'testTool'
      },
      translate: key => key,
      actions: {
        getProjectProgressForTools: () => jest.fn(),
        setProjectToolGL: () => jest.fn(),
        launchTool: () => jest.fn()
      },
      developerMode: false
    };
    const renderedValue =  renderer.create(
      <MuiThemeProvider>
        <ToolCard {...props} />
      </MuiThemeProvider>
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  test('Test GL Selection Change', () => {
    const props = {
      loggedInUser: true,
      currentProjectToolsProgress: {
        testTool: 0.5
      },
      currentProjectToolsSelectedGL: {
        testTool: 'en'
      },
      manifest: {
        project: {
          id: 'tit'
        }
      },
      metadata: {
        name: 'testTool'
      },
      translate: key => key,
      actions: {
        getProjectProgressForTools: () => jest.fn(),
        setProjectToolGL: () => jest.fn(),
        launchTool: () => jest.fn()
      },
      developerMode: false
    };
    const wrapper = shallow(<ToolCard {...props} />);
    const toolCard = wrapper.instance();
    expect(toolCard.state.selectedGL).toEqual(props.currentProjectToolsSelectedGL[props.metadata.name]);
    const newLanguageId = 'hi';
    toolCard.selectionChange(newLanguageId);
    expect(toolCard.state.selectedGL).toEqual(newLanguageId);
  });

  test('Test if no GL in props the default gateway language should be set', () => {
    const props = {
      loggedInUser: true,
      currentProjectToolsProgress: {
        testTool: 0.5
      },
      currentProjectToolsSelectedGL: {},
      manifest: {
        project: {
          id: 'tit'
        }
      },
      metadata: {
        name: 'testTool'
      },
      translate: key => key,
      actions: {
        getProjectProgressForTools: () => jest.fn(),
        setProjectToolGL: () => jest.fn(),
        launchTool: () => jest.fn()
      },
      developerMode: false
    };
    const wrapper = shallow(<ToolCard {...props} />);
    const toolCard = wrapper.instance();
    expect(toolCard.state.selectedGL).toEqual(DEFAULT_GATEWAY_LANGUAGE);
  });
});
