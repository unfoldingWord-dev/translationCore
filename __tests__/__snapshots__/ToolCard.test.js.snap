// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ToolCard component Comparing ToolCard Component render with snapshot taken 11/07/2017 in __snapshots__ should match 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "margin": "6px 0px 10px",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <img
      src={undefined}
      style={
        Object {
          "float": "left",
          "height": "90px",
          "margin": "10px",
        }
      }
    />
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "fontWeight": 500,
          "muiPrepared": true,
          "padding": 16,
          "position": "relative",
          "whiteSpace": "nowrap",
        }
      }
    >
      <div
        style={
          Object {
            "display": "inline-block",
            "muiPrepared": true,
            "paddingRight": "90px",
            "verticalAlign": "top",
            "whiteSpace": "normal",
          }
        }
      >
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fontSize": 15,
              "fontWeight": "bold",
              "muiPrepared": true,
            }
          }
        />
        <span
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "display": "block",
              "fontSize": 14,
              "muiPrepared": true,
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "float": "right",
          }
        }
      >
        <div>
          <table>
            <tbody>
              <tr>
                <th
                  style={
                    Object {
                      "padding": "5px",
                    }
                  }
                >
                  <span
                    className="glyphicon glyphicon-pencil"
                    style={
                      Object {
                        "fontSize": "18px",
                      }
                    }
                  />
                </th>
                <th
                  style={
                    Object {
                      "padding": "5px",
                    }
                  }
                >
                  <span
                    className="badge"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "borderColor": "#000",
                        "borderRadius": "6px",
                        "borderStyle": "solid",
                        "borderWidth": "1px",
                        "color": "#000",
                      }
                    }
                  >
                    0
                  </span>
                </th>
                <th
                  style={
                    Object {
                      "padding": "5px",
                    }
                  }
                >
                  <svg
                    height="18"
                    style={
                      Object {
                        "display": "inline-block",
                        "verticalAlign": "middle",
                      }
                    }
                    viewBox="0 0 475.082 475.082"
                    width="18"
                  >
                    <path
                      d="M107.067,317.195c1.713-1.708,2.568-3.898,2.568-6.563c0-2.663-0.855-4.853-2.568-6.571    c-1.714-1.707-3.905-2.562-6.567-2.562H9.135c-2.666,0-4.853,0.855-6.567,2.562C0.859,305.772,0,307.962,0,310.632    c0,2.665,0.855,4.855,2.568,6.563c1.714,1.711,3.905,2.566,6.567,2.566H100.5C103.166,319.766,105.353,318.91,107.067,317.195z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M310.629,109.634c2.669,0,4.859-0.855,6.563-2.568c1.718-1.711,2.574-3.901,2.574-6.567V9.138    c0-2.659-0.856-4.85-2.574-6.565c-1.704-1.711-3.895-2.57-6.563-2.57c-2.662,0-4.853,0.859-6.563,2.57    c-1.711,1.713-2.566,3.903-2.566,6.565v91.361c0,2.666,0.855,4.856,2.566,6.567C305.784,108.779,307.974,109.634,310.629,109.634z    "
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M118.771,347.184c-2.478,0-4.664,0.855-6.567,2.563l-73.089,73.087c-1.713,1.902-2.568,4.093-2.568,6.567    c0,2.474,0.855,4.664,2.568,6.566c2.096,1.708,4.283,2.57,6.567,2.57c2.475,0,4.665-0.862,6.567-2.57l73.089-73.087    c1.714-1.902,2.568-4.093,2.568-6.57c0-2.471-0.854-4.661-2.568-6.563C123.436,348.039,121.245,347.184,118.771,347.184z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M356.315,127.905c2.283,0,4.473-0.855,6.571-2.565l73.087-73.089c1.707-1.903,2.562-4.093,2.562-6.567    c0-2.475-0.855-4.665-2.562-6.567c-1.91-1.709-4.093-2.568-6.571-2.568c-2.471,0-4.66,0.859-6.563,2.568l-73.087,73.089    c-1.708,1.903-2.57,4.093-2.57,6.567c0,2.474,0.862,4.661,2.57,6.567C351.846,127.05,354.037,127.905,356.315,127.905z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M350.607,193.005c-4-3.999-9.328-7.994-15.988-11.991l-5.14,68.238l78.23,78.508c5.328,5.328,7.987,11.807,7.987,19.417    c0,7.423-2.662,13.802-7.987,19.13l-41.977,41.686c-5.146,5.146-11.608,7.666-19.417,7.566c-7.81-0.1-14.271-2.707-19.411-7.854    l-77.946-78.225l-68.234,5.144c3.999,6.656,7.993,11.988,11.991,15.985l95.362,95.643c15.803,16.18,35.207,24.27,58.238,24.27    c22.846,0,42.154-7.898,57.957-23.695l41.977-41.685c16.173-15.8,24.27-35.115,24.27-57.958c0-22.46-7.994-41.877-23.982-58.248    L350.607,193.005z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M472.518,157.889c-1.711-1.709-3.901-2.565-6.563-2.565h-91.365c-2.662,0-4.853,0.855-6.563,2.565    c-1.715,1.713-2.57,3.903-2.57,6.567c0,2.666,0.855,4.856,2.57,6.567c1.711,1.712,3.901,2.568,6.563,2.568h91.365    c2.662,0,4.853-0.856,6.563-2.568c1.708-1.711,2.563-3.901,2.563-6.567C475.082,161.792,474.226,159.602,472.518,157.889z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M109.348,67.099c5.523-5.14,11.991-7.705,19.417-7.705c7.611,0,14.084,2.663,19.414,7.993l77.943,78.227l68.234-5.142    c-4-6.661-7.99-11.991-11.991-15.987l-95.358-95.643c-15.798-16.178-35.212-24.27-58.242-24.27c-22.841,0-42.16,7.902-57.958,23.7    L28.837,69.955C12.659,85.756,4.57,105.073,4.57,127.912c0,22.463,7.996,41.877,23.982,58.245l95.93,95.93    c3.995,4.001,9.325,7.995,15.986,11.991l5.139-68.521L67.377,147.33c-5.327-5.33-7.992-11.801-7.992-19.417    c0-7.421,2.662-13.796,7.992-19.126L109.348,67.099z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                    <path
                      d="M164.454,365.451c-2.667,0-4.854,0.855-6.567,2.563c-1.711,1.711-2.568,3.901-2.568,6.57v91.358    c0,2.669,0.854,4.853,2.568,6.57c1.713,1.707,3.9,2.566,6.567,2.566c2.666,0,4.853-0.859,6.567-2.566    c1.713-1.718,2.568-3.901,2.568-6.57v-91.358c0-2.662-0.855-4.853-2.568-6.57C169.306,366.307,167.116,365.451,164.454,365.451z"
                      style={
                        Object {
                          "fill": undefined,
                        }
                      }
                    />
                  </svg>
                </th>
                <th
                  style={
                    Object {
                      "padding": "5px",
                    }
                  }
                >
                  <span
                    className="badge"
                    style={
                      Object {
                        "backgroundColor": "#fff",
                        "borderColor": "#000",
                        "borderRadius": "6px",
                        "borderStyle": "solid",
                        "borderWidth": "1px",
                        "color": "#000",
                      }
                    }
                  >
                    0
                  </span>
                </th>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
    <br />
    <ToolCardProgress
      progress={0.5}
    />
    <div />
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "space-between",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "fontSize": "18px",
              "padding": "10px 10px 0px",
            }
          }
        >
          <span>
            tools.see_more
          </span>
          <span
            className="glyphicon glyphicon-chevron-down"
            style={
              Object {
                "fontSize": "18px",
                "margin": "0px 0px 0px 6px",
              }
            }
          />
        </div>
      </div>
      <GlDropDownList
        bookID="tit"
        selectedGL="en"
        selectionChange={[Function]}
        translate={[Function]}
      />
      <span>
        <button
          className="btn-prime"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "margin": "10px",
              "width": "90px",
            }
          }
        >
          buttons.launch_button
        </button>
      </span>
    </div>
  </div>
</div>
`;
